obsidian.el

```diff
@@ -112,67 +128,165 @@ of `dirctory-files'."
 
 ;;;###autoload
 (defun obsidian-specify-path (&optional path)
-  "Specifies obsidian folder PATH to obsidian-folder variable.
+  "Sets `obsidian-directory' to PATH or user-selected directory.
 
 When run interactively asks user to specify the path."
   (interactive)
   (->> (or path (read-directory-name "Specify path to Obsidian folder"))
        (expand-file-name)
        (customize-set-value 'obsidian-directory)))
-(defvar obsidian--tags-list nil "List of Obsidian Notes tags generated by obsidian.el.")
 
+(define-minor-mode obsidian-mode
+  "Toggle minor `obsidian-mode' on and off.
+
+Interactively with no argument, this command toggles the mode.
+A positive prefix argument enables the mode, any other prefix
+argument disables it.  From Lisp, argument omitted or nil enables
+the mode, `toggle' toggles the state."
+  ;; The initial value.
+  :init-value nil
+  :lighter " obs"
+  :after-hook (obsidian-update)
+  :keymap (make-sparse-keymap))
+
+(defvar obsidian--debug-messages nil)
+
+;; TOOD: This doesn't produce the same tag rules as Obsidian Notes
 (defvar obsidian--tag-regex "#[[:alnum:]-_/+]+" "Regex pattern used to find tags in Obsidian files.")
 
 (defvar obsidian--basic-wikilink-regex "\\[\\[[[:graph:][:blank:]]*\\]\\]"
   "Regex pattern used to find wikilinks.")
+
 (defvar obsidian--basic-markdown-link-regex "\\[[[:graph:][:blank:]]+\\]\([[:graph:][:blank:]]*\)"
   "Regex pattern used to find markdown links.")
 
-(defvar obsidian--aliases-map (make-hash-table :test 'equal) "Alist of all Obsidian aliases.")
-
-(defun obsidian--clear-aliases-map ()
-  "Clears aliases map."
-  (interactive)
-  (setq obsidian--aliases-map (make-hash-table :test 'equal)))
+(defvar obsidian--vault-cache nil

```

Comment:
This is the new data structure for tracking obsidian files and their associated links, tags, and aliases